File: ws/Perez/U2/WS24/Zoo/src/main/java/ec/edu/espe/zoo/model/Monkey.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public class Monkey extends Primate{

    public Monkey(int id, String description, Date bornOnDate, char gender, Cage cage, int numberOfBones, boolean vetebrate) {
        super(id, description, bornOnDate, gender, cage, numberOfBones, vetebrate);
    }

    @Override
    public void feed(int food) {
        System.out.println("Feeding a monkey");
    }
    
}

File: hw/Pérez/U2/HW18ZooJavaScript/model/Monkey.js
import { Mammal } from './Mammal.js';

export class Monkey extends Mammal {
    constructor(id, description, bornOnDate, gender, cage, numberOfBones, vetebrate) {
        super(id, description, bornOnDate, gender, cage, numberOfBones, vetebrate);
    }

    feed(food) {
        console.log(`Feeding the monkey with ${food} units`);
    }

    toString() {
        return `Monkey{${super.toString()}}`;
    }
}

